pkg/kubectl/cmd/util/factory.go

```diff
@@ -57,6 +57,8 @@ type Factory struct {
 	Object func() (meta.RESTMapper, runtime.ObjectTyper)
 	// Returns a client for accessing Kubernetes resources or an error.
 	Client func() (*client.Client, error)
+	// Returns a client for accessing experimental Kubernetes resources or an error.
+	ExperimentalClient func() (*client.ExperimentalClient, error)
```

Comment:
How do you see this scaling out to handle addition API groups?  Separate methods for each?  Single method taking a string for `group` and returning back an interface that must be hard-cast?